scipy/special/tests/test_cdflib.py

```diff
@@ -844,3 +843,38 @@ def test_chndtr_edge_cases(self, x, df, nc, expected):
 @pytest.mark.parametrize("x", [0.1, 100])
 def test_chdtriv_p_equals_1_returns_0(x):
     assert sp.chdtriv(1, x) == 0
+
+
+class TestPdtrik:
+    @pytest.mark.parametrize("p, m, expected",
+                             [(0, 0.5, 0),
+                              (0.5, 0, 0),
+                              (0, 0, 0)])
+    def test_edge_cases(self, p, m, expected):
+        assert sp.pdtrik(p, m) == expected
+
+    @pytest.mark.parametrize("m", (0.1, 1, 10))
+    def test_p_equals_1_returns_nan(self, m):
+        assert np.isnan(sp.pdtrik(1, m))
+
+    def test_small_probabilities(self):
+        # Edge case: m = 0 or very small.
+        k = sp.pdtrik([[0], [0.25], [0.95]], [0, 1e-20, 1e-6])
+        assert_array_equal(k, np.zeros((3, 3)))
+
+    def test_roundtrip_against_pdtr(self):
+        m = [10, 50, 500]
+        k = 5
+        p = sp.pdtr(k, m)
+        assert_allclose(sp.pdtrik(p, m), k, atol=1.5e-7, rtol=0)
+
+    @pytest.mark.parametrize("p, m, k",
+                             [(1.8976107553682285e-40, 100, 2),
+                              (0.48670120172085135, 100, 99),
+                              (8.30383406699052e-69, 1000, 500),
+                              (2.252837804125894e-227, 100_000, 90_000)]
+                            )
```

Comment:
Style nit: this exact formatting case is covered in PEP 8:
```suggestion
                              (2.252837804125894e-227, 100_000, 90_000)])
```